### ISSUE TITLE
How to run model_main.py with multi-GPU?

### ISSUE THREAD

**Issue by kingstarcraft** [tensorflow/models#6047]
### System information
- **What is the top-level directory of the model you are using**:
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:windows 10
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:1.12
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:9.0/7.1
- **GPU model and memory**:GTX1080TI 11GB
- **Exact command to reproduce**: 

### Describe the problem
how to run model_main.py with multi-GPU

**Comment by rootkitchao** [comment]
You can use  legacy/train.py and set ''--num_clones=2 --ps_tasks=1''

**Comment by krustiv** [comment]
I set num_clones to my number of GPUs but on tensorboard, the classification and localization loss is for individual GPU. The total loss curve seems to be sum of all the losses for individual GPU. End up my loss is like 40% at the end of training! 
Is there something wrong in interpretation or something wrong in presentation of the loss curves?
Thank you

**Comment by royshil** [comment]
Have you seen #5421 ? 
Essentially you add to / modify in `model_main.py` the following lines
```
  strategy = tf.contrib.distribute.MirroredStrategy(num_gpus=2)
  config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir, train_distribute=strategy)
```

**Comment by Nyceane** [comment]
Getting following error, and it's 3-4 times slower than just running single GPU.  Although under nvidia-smi it shows Volatile-GPU-Util has some percentage on it.
```
W tensorflow/core/grappler/utils/graph_view.cc:830] No registered 'MultiDeviceIteratorGetNextFromShard' OpKernel for GPU devices compatible with node {{node MultiDeviceIteratorGetNextFromShard}}
	.  Registered:  device='CPU'

```

**Comment by mad-fogs** [comment]
> Have you seen #5421 ?
> Essentially you add to / modify in `model_main.py` the following lines
> 
> ```
>   strategy = tf.contrib.distribute.MirroredStrategy(num_gpus=2)
>   config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir, train_distribute=strategy)
> ```

Thanks, but the training is unbelievable slow...
